Jenkins reported testRoundRobinRouting failure

Jenkins reported testRoundRobinRouting failure:

```
18:08:32 FAIL: //heron/instance/tests/java:EmitDirectSpoutTest (see /var/lib/jenkins/.cache/bazel/_bazel_jenkins/8d6f21acdfe04ba486cfe7f3c8fbb82d/execroot/heron-mirror-release_5/bazel-out/local-opt/testlogs/heron/instance/tests/java/EmitDirectSpoutTest/test.log)
18:08:33 ____From Testing //heron/instance/tests/java:EmitDirectSpoutTest:
18:08:33 ==================== Test output for //heron/instance/tests/java:EmitDirectSpoutTest:
18:08:33 JUnit4 Test Runner
18:08:33 .Jul 26, 2017 1:08:32 AM com.twitter.heron.instance.Slave handleNewAssignment
18:08:33 INFO: Incarnating ourselves as test-spout with task id 0
18:08:33 Jul 26, 2017 1:08:32 AM com.twitter.heron.common.utils.misc.SerializeDeSerializeHelper getSerializer
18:08:33 WARNING: Serializer class name not provided. Fall back to Java serializer. This could cause serious performance degradation. You can specify to use Kryo as serializer. See https://twitter.github.io/heron/docs/developers/serialization/ for details
18:08:33 Jul 26, 2017 1:08:32 AM com.twitter.heron.instance.spout.SpoutInstance <init>
18:08:33 INFO: Enable Ack: false
18:08:33 Jul 26, 2017 1:08:32 AM com.twitter.heron.instance.spout.SpoutInstance <init>
18:08:33 INFO: EnableMessageTimeouts: true
18:08:33 Jul 26, 2017 1:08:32 AM com.twitter.heron.instance.Slave startInstance
18:08:33 INFO: Started instance.
18:08:33 Jul 26, 2017 1:08:32 AM com.twitter.heron.grouping.EmitDirectRoundRobinSpout emit
18:08:33 INFO: Emit direct tuple [A] to 0
18:08:33 Jul 26, 2017 1:08:32 AM com.twitter.heron.grouping.EmitDirectRoundRobinSpout emit
18:08:33 INFO: Emit direct tuple [B] to 1
18:08:33 Jul 26, 2017 1:08:32 AM com.twitter.heron.grouping.EmitDirectRoundRobinSpout emit
18:08:33 INFO: Emit direct tuple [A] to 2
18:08:33 Jul 26, 2017 1:08:32 AM com.twitter.heron.grouping.EmitDirectRoundRobinSpout emit
18:08:33 INFO: Emit direct tuple [B] to 3
18:08:33 EJul 26, 2017 1:08:32 AM com.twitter.heron.grouping.EmitDirectRoundRobinSpout emit
18:08:33 INFO: Emit direct tuple [A] to 4
18:08:33 Jul 26, 2017 1:08:32 AM com.twitter.heron.grouping.EmitDirectRoundRobinSpout emit
18:08:33 INFO: Emit direct tuple [B] to 5
18:08:33 Jul 26, 2017 1:08:32 AM com.twitter.heron.grouping.EmitDirectRoundRobinSpout emit
18:08:33 INFO: Emit direct tuple [A] to 6
18:08:33 
18:08:33 Time: 1.28
18:08:33 There was 1 failure:
18:08:33 1) testRoundRobinRouting(com.twitter.heron.grouping.EmitDirectSpoutTest)
18:08:33 java.lang.AssertionError: expected:<10> but was:<1>
18:08:33 	at org.junit.Assert.fail(Assert.java:88)
18:08:33 	at org.junit.Assert.failNotEquals(Assert.java:743)
18:08:33 	at org.junit.Assert.assertEquals(Assert.java:118)
18:08:33 	at org.junit.Assert.assertEquals(Assert.java:555)
18:08:33 	at org.junit.Assert.assertEquals(Assert.java:542)
18:08:33 	at com.twitter.heron.grouping.AbstractTupleRoutingTest$1.run(AbstractTupleRoutingTest.java:146)
18:08:33 	at com.twitter.heron.common.basics.WakeableLooper.executeTasksOnWakeup(WakeableLooper.java:136)
18:08:33 	at com.twitter.heron.common.basics.WakeableLooper.runOnce(WakeableLooper.java:75)
18:08:33 	at com.twitter.heron.common.basics.WakeableLooper.loop(WakeableLooper.java:65)
18:08:33 	at com.twitter.heron.grouping.AbstractTupleRoutingTest.testRoundRobinRouting(AbstractTupleRoutingTest.java:154)
18:08:33 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
18:08:33 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
18:08:33 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
18:08:33 	at java.lang.reflect.Method.invoke(Method.java:498)
18:08:33 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
18:08:33 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
18:08:33 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
18:08:33 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
18:08:33 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
18:08:33 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
18:08:33 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
18:08:33 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
18:08:33 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
18:08:33 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
18:08:33 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
18:08:33 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
18:08:33 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
18:08:33 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
18:08:33 	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
18:08:33 	at com.google.testing.junit.runner.junit4.CancellableRequestFactory$CancellableRunner.run(CancellableRequestFactory.java:89)
18:08:33 	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
18:08:33 	at org.junit.runner.JUnitCore.run(JUnitCore.java:138)
18:08:33 	at com.google.testing.junit.runner.junit4.JUnit4Runner.run(JUnit4Runner.java:114)
18:08:33 	at com.google.testing.junit.runner.BazelTestRunner.runTestsInSuite(BazelTestRunner.java:152)
18:08:33 	at com.google.testing.junit.runner.BazelTestRunner.main(BazelTestRunner.java:91)
18:08:33 
18:08:33 FAILURES!!!
18:08:33 Tests run: 1,  Failures: 1
18:08:33 
18:08:33 
18:08:33 BazelTestRunner exiting with a return value of 1
18:08:33 JVM shutdown hooks (if any) will run now.
18:08:33 The JVM will exit once they complete.
18:08:33 
18:08:33 -- JVM shutdown starting at 2017-07-26 01:08:32 --
18:08:33 
18:08:33 ================================================================================
```

--- a/heron/instance/tests/java/com/twitter/heron/grouping/AbstractTupleRoutingTest.java
+++ b/heron/instance/tests/java/com/twitter/heron/grouping/AbstractTupleRoutingTest.java
@@ -142,11 +142,11 @@ public abstract class AbstractTupleRoutingTest {
             assertEquals((Integer) tupleReceived, destTaskIds.get(0));
             tupleReceived++;
           }
-
-          assertEquals(expectedTuplesValidated, tupleReceived);
-          assertEquals(getExpectedComponentInitInfo(), groupingInitInfo.toString());
-          slaveTester.getTestLooper().exitLoop();
         }
+
+        assertEquals(expectedTuplesValidated, tupleReceived);
+        assertEquals(getExpectedComponentInitInfo(), groupingInitInfo.toString());
+        slaveTester.getTestLooper().exitLoop();
       }
     };
 

